lib/game.rb

```diff
@@ -0,0 +1,190 @@
+class Game
+
+  attr_accessor :board, :player_1, :player_2#, :cli
+
+  WIN_COMBINATIONS = [
+    [0, 1, 2],
+    [3, 4, 5],
+    [6, 7, 8],
+    [0, 3, 6],
+    [1, 4, 7],
+    [2, 5, 8],
+    [0, 4, 8],
+    [2, 4, 6]
+  ]
+
+  def initialize(player_1 = Players::Human.new("X"), player_2 = Players::Human.new("O"), board = Board.new)
+    self.player_1 = player_1
+    self.player_2 = player_2
+    self.board = board
+  end
+
+  def current_player
+    if self.board.turn_count % 2 == 0
+      return player_1 if player_1.token == "X"
+      return player_2
+    else
+      return player_2 if player_2.token == "O"
+      return player_1
+    end
+    # o_count = 0
+    # x_count = 0
+    # self.board.cells.each do |position|
+    #   if position == "X"
+    #     x_count += 1
+    #   elsif position == "O"
+    #     o_count += 1
+    #   end
+    # end
+    # # # assume player_1 goes first regardless of his token
+    # # if x_count == o_count
+    # #   return player_1
+    # # elsif x_count > o_count
+    # #   return player_2 if player_2.token == "O"
+    # # elsif o_count > x_count
+    # #   return player_2 if player_2.token == "X"
+    # # else
+    # #   # in a real project, this would be refactored to raise an exception
+    # #   return "Error! Do both players have the same token or did player 2 go before player 1?"
+    # # end
+    #
+    # # assume X goes first regardless of which player it is
+    # if x_count == o_count
+    #   return player_1 if player_1.token == "X"
+    #   return player_2 if player_2.token == "X"
+    # elsif x_count > o_count
+    #   return player_2 if player_2.token == "O"
+    #   return player_1 if player_1.token == "O"
+    # else
+    #   # in a real project, this would be refactored to raise an exception
+    #   return "Error! Do both players have the same token or did O go before X?"
+    # end
+  end
+
+  def win_array
+    WIN_COMBINATIONS.each do |win|
+      ref = self.board.cells[win[0]]
+      return [true, ref] if (ref == "X" || ref == "O") && self.board.cells[win[1]] == ref && self.board.cells[win[2]] == ref
+    end
+    [false, nil]
+  end
+
+  def won?
+    # WIN_COMBINATIONS.each do |win|
+    #   ref = self.board.cells[win[0]]
+    #   return true if (ref == "X" || ref == "O") && self.board.cells[win[1]] == ref && self.board.cells[win[2]] == ref
+    # end
+    # false
+    self.win_array[0]
```

Comment:
Actually, this was a deliberate refactoring. #won? and #winner were identical except for the return value so I implemented #win_array to keep things DRY.